app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java

```diff
@@ -0,0 +1,263 @@
+    public void deleteSubscriber(User user, Group group) {
+        int index = groups.indexOf(group);
+        groups.get(index).removeSubscriber(user);
+    }
+
+    @Override
+    public void createForum(User user, Group group, Forum forum) {
+        int index = groups.indexOf(group);
+        groups.get(index).addForum(forum);
+    }
+
+    @Override
+    public void deleteForum(User user, Group group, Forum forum) throws ForumNotFoundException, NotForumOwnerException {
+        if (!forumExists(group, forum)) {
+            throw new ForumNotFoundException();
+        }
+        if (!forum.getOwnerUsername().equals(user.getUsername())) {
+            throw new NotForumOwnerException();
+        }
+        group.removeForum(forum);
+    }
+
+    /**
+     * Checks whether a forum exists or not.
+     * @param group The group where the forum should be.
+     * @param forum The forum to check
+     * @return True if the forum was found or false otherwise
+     */
+    private boolean forumExists(Group group, Forum forum) {
+        for (Group g : groups) {
+            if (g.getName().equals(group.getName())) {
+                for (Forum f : g.getForums()) {
+                    if (f.getName().equals(forum.getName())) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void createPost(User user, Forum forum, Post post) throws ForumNotFoundException,
+        PostAlreadyExistingException {
+        if (!forumExists(forum.getGroup(), forum)) {
+            throw new ForumNotFoundException();
+        }
+        for (Group g : groups) {
+            if (g.getName().equals(forum.getGroup().getName())) {
+                for (Forum f : g.getForums()) {
+                    if (f.getName().equals(forum.getName())) {
+                        for (Post p : f.getPosts()) {
+                            if (p.getUsername().equals(post.getUsername())
+                                && p.getCreationDate().equals(post.getCreationDate())) {
+                                throw new PostAlreadyExistingException();
+                            }
+                        }
+                        f.addPost(post);
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void deletePost(User user, Forum forum, DateTime postCreationDate) throws ForumNotFoundException,
+        PostNotFoundException {
+        if (!forumExists(forum.getGroup(), forum)) {
+            throw new ForumNotFoundException();
+        }
+        for (Group g : groups) {
+            if (g.getName().equals(forum.getGroup().getName())) {
+                for (Forum f : g.getForums()) {
+                    if (f.getName().equals(forum.getName())) {
+                        boolean found = false;
+                        for (Post p : f.getPosts()) {
+                            if (p.getUsername().equals(user.getUsername())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Deeply nested if..then statements are hard to read](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5421176)